java/openjdk-java/portfile

```diff
@@ -0,0 +1,570 @@
+  
+        distname            openjfx-${version}-ea+9_osx-aarch64_bin-sdk
+    }
+    depends_lib          port:openjdk-18-jdk
+}
+subport openjdk-18-jdk {
+    version             18
+    description         openjdk 18 builds
+    set build           31
+    long description    ${long_description_openjdk-ea}
+    supported_archs     arm64 x86_64
+    if {${configure.build_arch} eq "x86_64"} {
+        master_sites        https://download.java.net/java/early_access/jdk${version}/${build}/GPL/
+  
+        checksum            sha256  9d733bc6dd0cc4a7eb84ae2134f1c32accd108c46d54414d01cb72da1415f052 \
+                            rmd160  503b8502b90864ba747366abc9a1bed5e46593d1 \
+                            size    185343149
+  
+        distname            openjdk-${version}-ea+{build}_macos-x64_bin
+    } elseif {${configure.build_arch} eq "x86_64"} {          
+        master_sites        https://download.java.net/java/early_access/jdk${version}/${build}/GPL/
+  
+        checksum            sha256  f3a32f4f5aa6d8a62bc979bedcad624b1fec1c3fd501430e8650b5724f26b4b3 \
+                            rmd160  2c38f775f9b323bda10d1e768482e1b0e804da85 \
+                            size    183200268
+  
+        distname            openjdk-${version}-ea+{build}_macos-aarch64_bin
+    }
+    worksrcdir          jdk-${version}.jdk
+}
+subport openjdk-19-jdk {
+    version             19
+    description         openjdk 19 builds
+    set build           5
+    long description    ${long_description_openjdk-ea}
+    supported_archs     arm64 x86_64
+    if {${configure.build_arch} eq "x86_64"} {
+        master_sites        https://download.java.net/java/early_access/jdk${version}/${build}/GPL/
+  
+        checksum            sha256  618a2cdf305924507e5142c1575e1c22d809c1f766f16793fd39aada38ac8e2c \
+                            rmd160  f3eac2953f06a73da5d90b7172c6c0a407e5e844 \
+                            size    186298743
+  
+        distname            openjdk-${version}-ea+{build}_macos-x64_bin
+    } elseif {${configure.build_arch} eq "arm64"} {          
+        master_sites        https://download.java.net/java/early_access/jdk${version}/${build}/GPL/
+  
+        checksum            sha256  d315456a64de458bf0bb9e8ad5c057dfcc111c0feb2414f2d9ba4bef276393a1 \
+                            rmd160  3aa40f031760d05df927771b3a8b75e4f78ad398 \
+                            size    184139469
+  
+        distname            openjdk-${version}-ea+{build}_macos-aarch64_bin
+    }
+    worksrcdir          jdk-${version}.jdk
+}
+
+if {[string match *-jdk ${subport}]} {
+    homepage            https://openjdk.java.net
+} elseif {[string match *-jfx ${subport}]} {
+    homepage            https://openjfx.io
+} elseif {[string match graalvm-* ${subport}]} {
+    homepage            https://www.graalvm.org
+}
+{
+    destroot.violate_mtree yes
+    set target /usr/local
+    set destroot_target ${destroot} ${target}
```

Comment:
why would we want to install anything is `/usr/local`; and secondly, are you sure you want to have a "space" between `${destroot}` and `${target}`?

Reply:
also please fill in the PR checklist above

Reply:
(see here :- https://wiki.openjdk.java.net/display/BSDPort/Darwin10Build and https://wiki.openjdk.java.net/display/BSDPort/Darwin9Build

Reply:
sure, these are general instructions on how to build stuff - MacPorts installs its software in `/opt/local/` or whatever custom prefix the end-user decided on. So I am pretty sure we don't want to install anything in `/usr/local`. 

The PR checklist is a list of things that should be all done before submitting a PR - so it seems you have a bit more work to do ;)

Reply:
Ok

Reply:
Changed /usr/local to /opt/local

Reply:
it should be `${prefix}` to accommodate custom prefixes a user might install MacPorts in. Also, when doing that there should be no need for `destroot.violate_mtree yes` as you are presumably not violating the correct installation location anymore. Please do test these things locally yourself, read the guide about Portfiles and/or check other examples.